### PR TITLE
feat: add email validation 

### PR DESCRIPTION
**Pull request by spaenleh**
This PR adds the code necessary to perform email validation without the need for an external library. The code is heavily based on the npm package `validator.js`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -302,3 +302,7 @@ export {
   type ChatBotMessage,
   GPTVersion,
 } from './chatbot/chatbot.js';
+
+export { isEmail } from './validation/isEmail.js';
+export { isFQDN } from './validation/isFQDN.js';
```
Comment (pyphilia) [inline src/index.ts:307]: What is it? Maybe use the full name?

**Review by pyphilia — APPROVED** [review]
Thanks for the PR, I have the same comment : add the source of the code.

File: src/index.ts (hunk 1 above)
Comment (spaenleh) [reply src/index.ts:307]: `FQDN` stands for Fully Qualified Domain Name. Not sure I should use the full ? What do you think ?

Comment (ReidyT) [reply src/index.ts:307]: I think it's okay to let it like that, this acronym is known enough in my sense. It'd be like using full name for HTTP.

**Review by ReidyT — APPROVED** [review]
LGTM 🌨️

**Pull request merged**